src/languages/fsharp.js

```diff
@@ -247,35 +247,43 @@ export default function(hljs) {
-  // const OPERATOR_CHAR_RE = /[!%&*+\-\/<=>@^|~?]/;
-  // The lone dot operator is special. It cannot be redefined, and we don't want to highlight it. It can be used as part of a multi-chars operator though.
-  const OPERATOR_CHAR_OR_DOT_RE = regex.either(OPERATOR_CHAR_RE, /\./);
-  // When a dot is present, it must be followed by another operator char:
-  const OPERATOR_FIRST_CHAR_OF_MULTIPLE_RE = regex.concat(OPERATOR_CHAR_OR_DOT_RE, regex.lookahead(OPERATOR_CHAR_OR_DOT_RE));
-  const SYMBOLIC_OPERATOR_RE = regex.either(
-    regex.concat(OPERATOR_FIRST_CHAR_OF_MULTIPLE_RE, OPERATOR_CHAR_OR_DOT_RE, '*'), // Matches at least 2 chars operators
-    regex.concat(OPERATOR_CHAR_RE, '+'), // Matches at least one char operators
-  );
-  const OPERATOR = {
-    scope: 'operator',
-    match: regex.either(
-      // symbolic operators:
-      SYMBOLIC_OPERATOR_RE,
-      // other symbolic keywords:
-      // Type casting and conversion operators:
-      /:\?>/,
-      /:\?/,
-      /:>/,
-      /:=/, // Reference cell assignment
-      /::?/, // : or ::
-      /\$/), // A single $ can be used as an operator
-    relevance: 0
-  };
+  const makeOperatorMode = function(includeEqual) {
+    // List or symbolic operator characters from the FSharp Spec 4.1, minus the dot, and with `?` added, used for nullable operators.
+    let allOperatorChars;
+    if (includeEqual)
+      allOperatorChars = "!%&*+-/<=>@^|~?";
+    else
+      allOperatorChars = "!%&*+-/<>@^|~?";
+    const OPERATOR_CHARS = Array.from(allOperatorChars);
+    const OPERATOR_CHAR_RE = regex.concat('[', ...OPERATOR_CHARS.map(regex.escape), ']');
+    // The lone dot operator is special. It cannot be redefined, and we don't want to highlight it. It can be used as part of a multi-chars operator though.
+    const OPERATOR_CHAR_OR_DOT_RE = regex.either(OPERATOR_CHAR_RE, /\./);
+    // When a dot is present, it must be followed by another operator char:
+    const OPERATOR_FIRST_CHAR_OF_MULTIPLE_RE = regex.concat(OPERATOR_CHAR_OR_DOT_RE, regex.lookahead(OPERATOR_CHAR_OR_DOT_RE));
+    const SYMBOLIC_OPERATOR_RE = regex.either(
+      regex.concat(OPERATOR_FIRST_CHAR_OF_MULTIPLE_RE, OPERATOR_CHAR_OR_DOT_RE, '*'), // Matches at least 2 chars operators
+      regex.concat(OPERATOR_CHAR_RE, '+'), // Matches at least one char operators
+    );
+    return {
+      scope: 'operator',
+      match: regex.either(
+        // symbolic operators:
+        SYMBOLIC_OPERATOR_RE,
+        // other symbolic keywords:
+        // Type casting and conversion operators:
+        /:\?>/,
+        /:\?/,
+        /:>/,
+        /:=/, // Reference cell assignment
+        /::?/, // : or ::
+        /\$/), // A single $ can be used as an operator
+      relevance: 0
+    };
+  }
+
+  const OPERATOR = makeOperatorMode(true);
+  // This variant is used when matching '=' should end a parent mode:
+  const OPERATOR_WITHOUT_EQUAL = makeOperatorMode(false);
```

Comment:
`true` and `false` have no meaning (in context) and as such are difficult to read... use named arguments/object literals instead:

```
const OPERATOR = makeOperatorMode({ includeEqual: false}); 
```

Reply:
Ok!

It's the first time I use destructuring like that, so let me know if I did something wrong.